File: api/src/main/java/com/gkh/syntheticmonitor/repository/SMTestRepository.java
package com.gkh.syntheticmonitor.repository;

import com.gkh.syntheticmonitor.model.SMTest;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Modifying;
import org.springframework.data.jpa.repository.Query;

import java.util.List;

public interface SMTestRepository extends JpaRepository<SMTest, String> {

	@Modifying
	@Query("update SMTest test set test.readyToExecute = true where test.active=true and " +
			"DATEDIFF(SECOND, test.timeLastExecuted, CURRENT_TIMESTAMP) >= test.scheduleTimeInSeconds")
	void updateReadyToExecute();

	@Query("select test from SMTest test where test.active=true and test.readyToExecute=true")
	List<SMTest> selectReadyToExecuteTests();

	@Query("select test from SMTest test")
	List<SMTest> selectTests();

}

File: api/src/main/kotlin/com/gkh/syntheticmonitor/repository/SMTestRepository.kt
package com.gkh.syntheticmonitor.repository

import com.gkh.syntheticmonitor.model.SMTest
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Modifying
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository

@Repository
interface SMTestRepository : JpaRepository<SMTest, String> {
    @Modifying
    @Query("update SMTest test set test.isReadyToExecute = true where test.isActive=true and " +
            "DATEDIFF(SECOND, test.timeLastExecuted, CURRENT_TIMESTAMP) >= test.scheduleTimeInSeconds")
    fun updateReadyToExecute()

    @Query("select test from SMTest test where test.isActive=true and test.isReadyToExecute=true")
    fun selectReadyToExecuteTests(): List<SMTest>

    @Query("select test from SMTest test")
    fun selectTests(): List<SMTest>
}